File: songs/go/05-map.go
package main

import (
	"example.com/mysongs"
	"fmt"
)

func mapAndFilter(songs []mysongs.Song) [][]string {
	var tagss [][]string

	for _, song := range songs {
		if song.Tags != nil {
			tagss = append(tagss, song.Tags)
		}
	}

	return tagss
}

func main() {
	tagss := mapAndFilter(mysongs.GetSongs())
	fmt.Println(tagss)
}

File: songs/rust/05-map.rs
mod my_songs;

fn main() {
    let songs: Vec<my_songs::Song> = my_songs::get_songs();

    let tags_os: Vec<Option<Vec<&str>>> =
        songs.into_iter().map(|song| song.tags).collect();
    
    for tags_o in tags_os.iter() {
        println!("{:?}", tags_o);
    }
}
